modules/elasticsearch_helper_views/src/Plugin/views/query/Elasticsearch.php

```diff
@@ -539,12 +549,13 @@ public function getCacheContexts() {
       $contexts = Cache::mergeContexts($contexts, $query_builder->getCacheContexts());
     }
 
-    // Add base entity type cache context.
-    if (isset($this->options['entity_relationship']['entity_type_key'])) {
-      $entity_type_id = $this->getNestedValue($this->options['entity_relationship']['entity_type_key'], []);
+    foreach ($this->getEntityRelationships() as $relationship) {
+      if (isset($relationship['entity_type_key'])) {
+        $entity_type_id = $this->getNestedValue($relationship['entity_type_key'], []);
 
-      if ($entity_type = \Drupal::entityManager()->getDefinition($entity_type_id, FALSE)) {
-        $contexts = Cache::mergeContexts($contexts, $entity_type->getListCacheContexts());
+        if ($entity_type = \Drupal::entityManager()->getDefinition($entity_type_id, FALSE)) {
```

Comment:
Can existing $this->entityTypeManager be reused for this?

Reply:
Good point.